src/models/hooks/activityReportObjective.js

```diff
@@ -34,6 +36,10 @@ const recalculateOnAR = async (sequelize, instance, options) => {
   `, { transaction: options.transaction });
 };
 
+const beforeValidate = async (sequelize, instance, options) => {
+  validateChangedOrSetEnums(sequelize, instance);
```

Comment:
If this throws an exception where do we bubble up to from a hook?

Reply:
the very top, the operation will fail

Reply:
So rollback of the current operation. How does this look on the FE, does the user know what happened in terms of a invalid enum.

Reply:
the issue this is directly solving was empty string was trying to be saved. Which is not valid. This will turn empty string into null and save that if valid.

Reply:
It just also now checks for all other valid values in the same pass. So that the error message is more descriptive.